### PR TITLE
remove proxy core::git::Oid

### PR DESCRIPTION
**Pull request by krlvi**

### CODE REVIEW

**Comment by Caleb-T-Owens** [comment]
Heck yeah!

Oid.into() into the trash!

**Comment by krlvi** [comment]
> Heck yeah!
> 
> Oid.into() into the trash!

;)

File: crates/gitbutler-core/src/git/diff.rs [hunk 1]
```diff
@@ -124,7 +123,11 @@ impl GitHunk {
 #[serde(rename_all = "camelCase")]
 pub struct HunkLock {
     pub branch_id: Id<Branch>,
-    pub commit_id: git::Oid,
+    #[serde(
+        serialize_with = "crate::serde::oid::serialize",
+        deserialize_with = "crate::serde::oid::deserialize"
```
Comment (Caleb-T-Owens) [inline crates/gitbutler-core/src/git/diff.rs]:
```suggestion
with = "crate::serde::oid"
```
https://serde.rs/field-attrs.html#with

Comment (krlvi) [reply crates/gitbutler-core/src/git/diff.rs]: thanks!

**Comment by krlvi** [comment]
seems like its failing parsing toml... hmm
```
 gitbutler_core::fs::read_toml_file_or_default
             at ./src/fs.rs:116:9
```

```
Caused by:
    TOML parse error at line 9, column 1
      |
    9 | [branches.a2de6835-0594-4728-a193-3f1d4923fe81]
      | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    missing field `upstream_head`
    ```

**Comment by krlvi** [comment]
thank you @Byron <3

**Comment by Byron** [comment]
Thank *you* for these refactors! So much better now!

**Pull request merged**